src/credentials/loginManager.ts

```diff
@@ -40,12 +41,28 @@ export class LoginManager {
                 throw new Error(`No credentials found for id ${asString(args.providerId)}`)
             }
 
+            // In V2 of the AWS JS SDK, the STS client tries to parse from the credentials file even
+            // if it doesn't exist, throwing an uncaught exception.
+            // This is because it doesn't check for the region we passed to it...
+            // Instead we have to set an enviromental variable
+            // TODO: Remove this hack when the toolkit migrates to version 3 of the SDK
+
             const credentialsRegion = provider.getDefaultRegion() ?? this.defaultCredentialsRegion
+            const env = process.env as EnvironmentVariables
+            const tmp = env.AWS_REGION
+            env.AWS_REGION = env.AWS_REGION ?? credentialsRegion
+
             const accountId = await getAccountId(storedCredentials.credentials, credentialsRegion)
```

Comment:
Let's create a `getAccountId2()` function that encapsulates this workaround. That makes the scope of the workaround very obvious, and  easy to remove workaround later.
 
Is hack also needed in `promptAndCreateNewCredentialsFile()`, which also uses `getAccountId()`?

Reply:
Seems like it's needed, though I was unable to create a scenario that would trigger the bug.